### ISSUE TITLE
Initial version for resolver for supporting DataContract and DataMember attributes in System.Text.Json

### ISSUE THREAD

**Issue by zcsizmadia** [dotnet/runtime#75226]
System.Text.Json serializer does not support DataContact and DataMember attributes prior .NET 7.0. System.Text.Json v7.x+ supports custom type resolvers. This library adds support for DataContract and DataMember attributes using a custom resolver.

https://github.com/zcsizmadia/ZCS.DataContractResolver

The library has many test cases for DataContract, which are serialized with System.Text.Json and Newtonsoft.Json and the serialized json string results must match.

**Comment by krwq** [comment]
Good idea :-) one thing to consider is to perhaps provide implementation as modifier which can be passed `DefaultTypeInfoResolver.Modifiers` (which would also allow to serialize other types).

Since this issue is in dotnet/runtime, what output do you expect from this issue? Should we treat this as an FYI announcement (something we can point to other people asking for this) or more like something you'd like added into JSON?

**Comment by zcsizmadia** [comment]
At this point it was more like an announcement, however considering the fact that for many people this is a major pain point for switching to STJ and the fact that the DataContract and DataMember attributes are part of the .NET runtime libraries, this feature could be part of STJ. E.g as `System.Text.Json.Serialization.Metadata.DataContract` library. If there is interest for that, this can be turned into a feature request.

The code itself is pretty much inspired by [63686](https://github.com/dotnet/runtime/issues/63686), the ony real addition is to make sure to match the serialization output of Newtonsoft.Json
